File: app/channels/online_channel.rb
class OnlineChannel < ApplicationCable::Channel
  def subscribed
    stream_from "online_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def appear
    ActionCable.server.broadcast "online_channel", message: "#{current_user.username}"
  end
end

File: app/javascript/channels/online_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("OnlineChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    this.perform("appear")
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    console.log(data.message)
    $('#user_table').append('<a class="item">' + data.message + '</a>')
  }
});
